packages/injective-cosmwasm/src/types.rs

```diff
@@ -130,19 +139,25 @@ impl<'de> Deserialize<'de> for SubaccountId {
     where
         D: Deserializer<'de>,
     {
-        let s = String::deserialize(deserializer)?;
+        let subaccount_id = String::deserialize(deserializer)?;
 
-        if !s.starts_with("0x") {
+        let is_nonce_derived = subaccount_id.len() <= MAX_SUBACCOUNT_NONE_LENGTH;
+        if is_nonce_derived && subaccount_id.parse::<u64>().is_err() {
+            let error_message = "Invalid nonce: when subaccount_id is derived from nonce, it must be a valid u64";
+            return Err(D::Error::custom(error_message));
+        }
+
+        if !subaccount_id.starts_with("0x") {
```

Comment:
wait, am I missing something? if it is `is_nonce_derived` and a valid `u64` it will fail this check, won't it? and then then ext one which checks the length.

Also, not every `u64` is valid, it has to be < 999.

Also... `""` is also a valid one (meaning the default one).

Reply:
Ah indeed, fixed. As for 'it has to be < 999' that is enforced by checking if <= MAX_SUBACCOUNT_NONE_LENGTH.